### ISSUE TITLE
Uncaught TypeError: Cannot read properties of null (reading 'appendChild')

### ISSUE THREAD

**Issue by JPustkuchen** [jfeltkamp/cookiesjsr#36]
After upgrading to 2.0.2 now I have the following error exactly where I had https://github.com/jfeltkamp/cookiesjsr/issues/35 before.
Chrome on Windows 11 when reloading the page without having cookies accepted or not.

The banner isn't showing up, instead the following error appears in console:
```
cookiesjsr.min.js?v=10.3.10:12 Uncaught TypeError: Cannot read properties of null (reading 'appendChild')
    at cookiesjsr.min.js?v=10.3.10:12:17416
    at Pn (cookiesjsr.min.js?v=10.3.10:12:10450)
    at ze (cookiesjsr.min.js?v=10.3.10:12:11140)
    at Pe (cookiesjsr.min.js?v=10.3.10:12:6630)
    at Ir (cookiesjsr.min.js?v=10.3.10:12:7163)
    at si (cookiesjsr.min.js?v=10.3.10:12:17397)
    at ii (cookiesjsr.min.js?v=10.3.10:12:17023)
    at cookiesjsr.min.js?v=10.3.10:12:41855
```

This is where it happens:
```
a(Nt(Mn)),
    jt.add(a);
    var l = void 0
      , f = Ir( () => {
        var c = n ?? t.appendChild(ft()); // <<<<<<<<<<<<< Uncaught TypeError: Cannot read properties of null (reading 'appendChild')
        return je( () => {
            if (s) {
```

I sent @jfeltkamp an email with the live project where it can be reproduced.

**Comment by JPustkuchen** [comment]
I was just thinking if it might be related to race conditions if the script is cached (and therefore loading very fast). Because it only happens on F5 but not on Ctrl+F5 (reload with cache refresh).

In Drupal we're using "defer" attribute. Using "async" might have been a reason, but defer is documented like this:

> If the defer attribute is set, it specifies that the script is downloaded in parallel to parsing the page, and executed after the page has finished parsing.

and I think cookiesjsr should work flawlessly with `defer` though?

Still looks like some kind of race-condition to me...

**Comment by jfeltkamp** [comment]
@JPustkuchen  Yes, you are right. The library is loading async in the header (on your example page) and is executed after JS file download finished. The target tag for cookies is at the bottom. 
I have no appendChild in my custom code. This must be Svelte code that injects the COOKiES frontend into the DOM.

The library in our COOKiES module already sets defer attribute, so wondering why it comes out as async in the header.  

Think, nothing to do here. :-)

**Comment by JPustkuchen** [comment]
Thanks @jfeltkamp! I'll take a look, as that page simply uses the last COOKiES version cookies 2.0.0-alpha1, no customizations.
I also verified alpha1 uses `defer`, not async and it does...
I'll take a deeper look now.

And indeed you're right! The output is
```
<script src="/modules/contrib/cookies/js/cookiesjsr.conf.js?v=10.3.10" async></script>
<script src="/libraries/cookiesjsr/dist/cookiesjsr.min.js?v=10.3.10" async></script>
```
Crazy...

**Comment by JPustkuchen** [comment]
This is really crazy. I looked for "async" in other modules used, and saw that we have reCaptcha installed, but don't use it anymore. So I uninstalled it and now the `defer` is back and `async` is gone.

Before that I cleared caches with no effect.

So either it's a side-effect of uninstalling something or recaptcha module broke the defer and made it async, which shouldn't happen...

I won't invest more time here now, but that's really unexpected. Thank you very much once again @jfeltkamp and sorry I didn't look into the HTML output as I would never have expected it to be wrong like this -.-

Now we at least know that `defer` is right and `async` can break things!
